File: data/java/tests/test_Schema_And.java
package com.example;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class test_Schema_And {
    private function_Schema_And validator;

    @BeforeEach
    void setUp() {
        validator = new function_Schema_And();
    }

    @Test
    void testValidData() {
        String result = validator.validate_user_input("admin", "securepassword123");
        assertEquals("Validation passed: username = admin, password = securepassword123", result);
    }

    @Test
    void testMissingUsername() {
        String result = validator.validate_user_input("", "securepassword123");
        assertEquals("Validation failed", result);
    }

    @Test
    void testShortPassword() {
        String result = validator.validate_user_input("admin", "short");
        assertEquals("Validation failed", result);
    }

    @Test
    void testNonStringUsername() {
        String result = validator.validate_user_input(null, "securepassword123");
        assertEquals("Validation failed", result);
    }

    @Test
    void testValidDataNoErrors() {
        String result = validator.validate_user_input("admin", "securepass123");
        assertEquals("Validation passed: username = admin, password = securepass123", result);
    }
}

File: data/cpp/tests/test_Schema_And.cpp
#include <gtest/gtest.h>
#include "../src/function_Schema_And.cpp"

class TestStringValidator : public ::testing::Test {
protected:
    void SetUp() override {
        validator = new StringValidator();
    }

    void TearDown() override {
        delete validator;
    }

    StringValidator* validator;
};

TEST_F(TestStringValidator, ValidData) {
    EXPECT_EQ(validator->validate_user_input("admin", "securepassword123"),
              "Validation passed: username = admin, password = securepassword123");
}

TEST_F(TestStringValidator, MissingUsername) {
    EXPECT_EQ(validator->validate_user_input("", "securepassword123"),
              "Validation failed");
}

TEST_F(TestStringValidator, ShortPassword) {
    EXPECT_EQ(validator->validate_user_input("admin", "short"),
              "Validation failed");
}

TEST_F(TestStringValidator, NonStringUsername) {
    EXPECT_EQ(validator->validate_user_input(12345, "securepassword123"),
              "Validation failed");
}

TEST_F(TestStringValidator, ValidDataNoErrors) {
    EXPECT_EQ(validator->validate_user_input("admin", "securepass123"),
              "Validation passed: username = admin, password = securepass123");
}

int main(int argc, char** argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}